File: php/src/Structural/Composite/OneTeaBag.php
<?

#//OneTeaBag.java - one composite extension - the "leaf"

require_once 'TeaBags.php';


class OneTeaBag extends TeaBags
{

function countTeaBags() { return 1; } 
function add(TeaBags $tb) { return 0; }
function remove(TeaBags $tb){ return 0; }

}

File: csharp/src/Structural/Composite/OneTeaBag.cs
//OneTeaBag.java - one composite extension - the "leaf"

namespace Structural.Composite {


public class OneTeaBag : TeaBags { 
    public OneTeaBag(string nameIn) {
        this.setName(nameIn);
    }
    
    override public int countTeaBags() {
        return 1;
    }
   
    override public bool add(TeaBags teaBagsToAdd) {
        return false;
    }
    override public bool remove(TeaBags teaBagsToRemove) {
        return false;
    }
}

}
